scaleway/helpers_k8s.go

```diff
@@ -0,0 +1,103 @@
+package scaleway
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	k8s "github.com/scaleway/scaleway-sdk-go/api/k8s/v1beta3"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type KubeconfigStruct struct {
+	ApiVersion string `yaml:"apiVersion"`
+	Clusters   []struct {
+		Name    string `yaml:"name"`
+		Cluster struct {
+			CertificateAuthorityData string `yaml:"certificate-authority-data"`
+			Server                   string `yaml:"server"`
+		} `yaml:"cluster"`
+	} `yaml:"clusters"`
+	Contexts []struct {
+		Name    string `yaml:"name"`
+		Context struct {
+			Cluster string `yaml:"cluster"`
+			User    string `yaml:"user"`
+		} `yaml:"context"`
+	} `yaml:"contexts"`
+	Kind  string `yaml:"kind"`
+	Users []struct {
+		Name string `yaml:"name"`
+		User struct {
+			Token string `yaml:"token"`
+		} `yaml:"user"`
+	} `yaml:"users"`
+}
+
+const (
+	K8SClusterWaitForReadyTimeout   = 10 * time.Minute
+	K8SClusterWaitForDeletedTimeout = 10 * time.Minute
+)
+
+func getK8SAPIWithRegion(d *schema.ResourceData, m interface{}) (*k8s.API, scw.Region, error) {
+	meta := m.(*Meta)
+	k8sAPI := k8s.NewAPI(meta.scwClient)
+
+	region, err := getRegion(d, meta)
+
+	return k8sAPI, region, err
+}
+
+func getK8SAPIWithRegionAndID(m interface{}, id string) (*k8s.API, scw.Region, string, error) {
+	meta := m.(*Meta)
+	k8sAPI := k8s.NewAPI(meta.scwClient)
+
+	region, ID, err := parseRegionalID(id)
+	return k8sAPI, region, ID, err
+}
+
+func waitK8SClusterReady(k8sAPI *k8s.API, region scw.Region, clusterID string) error {
+	now := time.Now()
+	for {
+		response, err := k8sAPI.GetCluster(&k8s.GetClusterRequest{
+			Region:    region,
+			ClusterID: clusterID,
+		})
+		if err != nil {
+			return err
+		}
+
+		if response.Status == k8s.ClusterStatusReady {
+			return nil
+		}
+		time.Sleep(time.Second * 5)
```

Comment:
When moving this method to the SDK, use [async.WaitSync](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/instance/v1/server_utils.go#L67)